core/src/main/java/com/linecorp/armeria/common/HttpRequestDuplicator.java

```diff
@@ -16,128 +16,51 @@
-
-    /**
-     * Creates a new instance wrapping an {@link HttpRequest} and publishing to multiple subscribers.
-     * The length of request is limited by default with the server-side parameter which is
-     * {@link Flags#defaultMaxResponseLength()}. If you are at client-side, you need to use
-     * {@link #HttpRequestDuplicator(HttpRequest, long)} and the {@code long} value should be greater than
-     * the length of request or {@code 0} which disables the limit.
-     * @param req the request that will publish data to subscribers
-     */
-    public HttpRequestDuplicator(HttpRequest req) {
-        this(req, Flags.defaultMaxRequestLength());
-    }
-
-    /**
-     * Creates a new instance wrapping an {@link HttpRequest} and publishing to multiple subscribers.
-     * @param req the request that will publish data to subscribers
-     * @param maxSignalLength the maximum length of signals. {@code 0} disables the length limit
-     */
-    public HttpRequestDuplicator(HttpRequest req, long maxSignalLength) {
-        this(req, maxSignalLength, null);
-    }
+public interface HttpRequestDuplicator extends StreamMessageDuplicator<HttpObject> {
 
     /**
-     * Creates a new instance wrapping an {@link HttpRequest} and publishing to multiple subscribers.
-     * @param req the request that will publish data to subscribers
-     * @param maxSignalLength the maximum length of signals. {@code 0} disables the length limit
-     * @param executor the executor to use for upstream signals.
+     * Returns a new {@link HttpRequest} that publishes the same elements with the {@link HttpRequest}
+     * that this duplicator is created from.
      */
-    public HttpRequestDuplicator(HttpRequest req, long maxSignalLength, @Nullable EventExecutor executor) {
-        super(requireNonNull(req, "req"), obj -> {
-            if (obj instanceof HttpData) {
-                return ((HttpData) obj).length();
-            }
-            return 0;
-        }, executor, maxSignalLength);
-        headers = req.headers();
-    }
-
-    @Override
-    public HttpRequest duplicateStream() {
-        return duplicateStream(headers);
-    }
-
     @Override
-    public HttpRequest duplicateStream(boolean lastStream) {
-        return duplicateStream(headers, lastStream);
-    }
-
-    /**
-     * Creates a new {@link HttpRequest} instance that publishes data from the {@code publisher} you create
-     * this factory with.
-     */
-    public HttpRequest duplicateStream(RequestHeaders newHeaders) {
-        return duplicateStream(newHeaders, false);
-    }
+    HttpRequest duplicate();
 
     /**
-     * Creates a new {@link HttpRequest} instance that publishes data from the {@code publisher} you create
-     * this factory with. If you specify the {@code lastStream} as {@code true}, it will prevent further
-     * creation of duplicate stream.
+     * Returns a new {@link HttpRequest} with the specified {@link RequestHeaders} that publishes the same
```

Comment:
Related to above it's not clear what it means to have specified headers but publish same objects

Reply:
How about this?
```java
* Returns a new {@link HttpRequest} with the specified {@link RequestHeaders} that publishes the same
* {@link HttpData}s and {@linkplain HttpHeaders trailers} with the {@link HttpRequest} that
* this duplicator is created from.
```

Reply:
Yeah I think that's clearer and nice, by the way `with the` -> `as the`

Reply:
Oops Thank!